R/db.R

```diff
@@ -95,9 +105,8 @@ dbWriteTableUntilSuccess <- function(dbconn, tblname, df, append = FALSE, row.na
   }
 
   repeat {
-    class_map <- list(integer = 'bigint', numeric = 'double precision', factor = 'text',
-                      double = 'double precision', character = 'text', logical = 'text')
-    field_types <- sapply(sapply(df, class), function(klass) class_map[[klass]])
+    field_classes <- vapply(df, function(col) class(col)[1L], character(1))
+    field_types <- vapply(field_classes, function(klass) CLASS_MAP[[klass]], character(1))
```

Comment:
At first I was about to say `vapply(field_classes, flip [[, character(1), CLASS_MAP)` and then I derped this isn't Haskell.